blosc/frame.c

```diff
@@ -1628,9 +1824,31 @@ void* frame_append_chunk(blosc2_frame* frame, void* chunk, blosc2_schunk* schunk
     memcpy(framep + header_len + cbytes, chunk, (size_t)cbytes_chunk);
     /* Copy the offsets */
     memcpy(framep + header_len + new_cbytes, off_chunk, (size_t)new_off_cbytes);
-  } else {
+  }
+  else if (frame->eframe) {
+    //Create nchunks.chunk file
+    eframe_append_chunk(frame, chunk, nchunks, cbytes_chunk);
+    // fileframe
+    char* eframe_name = malloc(strlen(frame->urlpath) + strlen("/chunks.b2frame") + 1);
+    sprintf(eframe_name, "%s/chunks.b2frame", frame->urlpath);
+    fp = fopen(eframe_name, "rb+");
+    free(eframe_name);
+    fseek(fp, header_len, SEEK_SET);
+    size_t wbytes = fwrite(off_chunk, 1, (size_t)new_off_cbytes, fp);  // the new offsets
+    fclose(fp);
+    if (wbytes != (size_t)new_off_cbytes) {
+      BLOSC_TRACE_ERROR("cannot write the offsets to fileframe.");
+      return NULL;
+    }
+    // Invalidate the cache for chunk offsets
+    if (frame->coffsets != NULL) {
```

Comment:
Move it out of `if` statement. In the `else` it must also be removed, thus we avoid duplicating code.